app/src/main/java/timur/gilfanov/messenger/MainActivity.kt

```diff
@@ -35,40 +31,50 @@ class MainActivity : ComponentActivity() {
         setContent {
             MessengerTheme {
                 val backStack = rememberNavBackStack(ChatList)
-                val currentUserId = ParticipantId(
-                    UUID.fromString("550e8400-e29b-41d4-a716-446655440000"),
-                )
+                val currentUserId = "550e8400-e29b-41d4-a716-446655440000".toParticipantId()
+
+                val chatNavDisplay = remember(backStack, currentUserId) {
```

Comment:
`backStack` is observable mutable list and reference will be the same. Compose compare this by reference equality and get notifications when list changes. So, no need in `backStack` key. `currentUserId` is immutable, so no need to use it a key either: 
```suggestion
                val chatNavDisplay = remember {
```